### PR TITLE
Features/jsmn 83 chat data

### PR DESCRIPTION
**Pull request by SophiaH67**

### CODE REVIEW

File: EasyIntern-Backend/Controllers/ChatController.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  public class ChatController : Controller
+  {
+    private readonly Context _context;
+
+    public ChatController(Context context)
+    {
+      _context = context;
+    }
+
+    [HttpGet("<id:int>")]
```
Comment (SilGosker) [inline EasyIntern-Backend/Controllers/ChatController.cs]: should be [HttpGet("{id:int}")

File: EasyIntern-Backend/Controllers/ChatController.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        .Include(e => e.ChatMessages)
+        .Include(e => e.Student)
+        .Include(e => e.Company)
+        .SingleOrDefaultAsync(e => e.Id == id && (e.CompanyId == userId || e.StudentId == userId));
+      if (chat == null)
+      {
+        return NotFound();
+      }
+      return Json(chat);
+    }
```
Comment (SilGosker) [inline EasyIntern-Backend/Controllers/ChatController.cs]: Looks like the function name should be Details?

File: EasyIntern-Backend/Controllers/ChatController.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      int userId = User.Id();
+      var chats = await _context.Chats.AsNoTracking()
+        .Include(e => e.ChatMessages)
+        .Include(e => e.Student)
+        .Include(e => e.Company)
+        .Where(e => e.CompanyId == userId || e.StudentId == userId).ToListAsync();
+      return Json(chats);
+    }
+
+    [HttpPost("<id:int>/message")]
```
Comment (SilGosker) [inline EasyIntern-Backend/Controllers/ChatController.cs]: Same as above, should be [HttpPost("{id:int}/message")]

File: EasyIntern-Backend/Controllers/ChatController.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      }
+      return Json(chat);
+    }
+
+    [HttpGet("")]
+    public async Task<IActionResult> Index()
+    {
+      int userId = User.Id();
+      var chats = await _context.Chats.AsNoTracking()
+        .Include(e => e.ChatMessages)
```
Comment (SilGosker) [inline EasyIntern-Backend/Controllers/ChatController.cs:42]: Should we include the chatmessages at the index page?

**Review by SilGosker — CHANGES_REQUESTED** [review]

File: EasyIntern-Backend/Controllers/ChatController.cs (hunk 4 above)
Comment (SophiaH67) [reply EasyIntern-Backend/Controllers/ChatController.cs:42]: Thought about doing a get request to `/chat/<id>/messages`, but in the end, when in the app do we want to list all chats without the messages? (I'm assuming  that in the index we want to show the latest chat message as a preview)

**Review by SilGosker — APPROVED** [review]

**Pull request merged**